pkg/controllers/management/kontainerdrivermetadata/data_getter.go

```diff
@@ -45,41 +44,21 @@ func GetRKEWindowsSystemImages(k8sVersion string, sysImageLister v3.RKEK8sWindow
 	return sysImage.SystemImages, err
 }
 
-func GetRKEAddonTemplate(k8sVersion string, addonName string, addonLister v3.RKEAddonLister, addons v3.RKEAddonInterface) (string, error) {
-	names := []string{
-		fmt.Sprintf("%s-%s", strings.ToLower(addonName), k8sVersion),
-		fmt.Sprintf("%s-%s", strings.ToLower(addonName), util.GetTagMajorVersion(k8sVersion)),
-		fmt.Sprintf("%s-%s", strings.ToLower(addonName), "default"),
-	}
-
-	for _, name := range names {
-		addon, err := addonLister.Get(namespace.GlobalNamespace, name)
-		if err != nil {
-			if !errors.IsNotFound(err) {
-				return "", err
-			}
-			continue
+func GetRKEAddonTemplate(addonName string, addonLister v3.RKEAddonLister, addons v3.RKEAddonInterface) (string, error) {
+	addon, err := addonLister.Get(namespace.GlobalNamespace, addonName)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return "", err
 		}
-		if addon.Labels[sendRKELabel] == "false" {
-			return "", nil
+		addon, err = addons.GetNamespaced(namespace.GlobalNamespace, addonName, metav1.GetOptions{})
+		if !errors.IsNotFound(err) {
+			return "", err
```

Comment:
We need to handle if err == errors.IsNotFound? Else we will hit panic below

Reply:
updated to return all errs if second Get fails